packages/cmf/src/assert.js

```diff
@@ -0,0 +1,22 @@
+// eslint-disable-next-line import/prefer-default-export
+export function assertTypeOf(options, attr, type, toThrow = true) {
+	if (type === 'Array' && options[attr] && Array.isArray(options[attr])) {
```

Comment:
1. In any case, your most determinant check is an undefined value.
Your first line should be 
```
if (options[attr] === undefined) {
    isTypeCorrect = true;
}
```
Then you can drop this check.

2. The multiple type mode is not really efficient, it test every `type`, but what you want to do is check if the value is one of the types.
`isTypeCorrect  = type.some(currentType => assertTypeOf(options, attr, currentType, false));`


```
export function assertTypeOf(options, attr, type, toThrow = true) {
	let isTypeCorrect;
	const value = options[attr];

	if (value === undefined) {
		isTypeCorrect = true;
	} else if (Array.isArray(type)) {
		isTypeCorrect  = type.some(currentType => assertTypeOf(options, attr, currentType, false));
	} else if (type === 'Array') {
		isTypeCorrect = Array.isArray(value);
	} else {
		isTypeCorrect = typeof options[attr] === type;
	}

	...
}
```

Reply:
updated but with a little bit different implementation